src/contrib/cluster/Akka.DistributedData/DeltaPropagationSelector.cs

```diff
@@ -105,19 +104,17 @@ public void Delete(string key)
                             var cacheKey = (key, fromSeqNr, toSeqNr);
                             if (!cache.TryGetValue(cacheKey, out var deltaGroup))
                             {
-                                using (var e = deltaEntriesAfterJ.Values.GetEnumerator())
+                                deltaGroup = deltaEntriesAfterJ.Values.Aggregate((d1, d2) =>
                                 {
-                                    e.MoveNext();
-                                    deltaGroup = e.Current;
-                                    while (e.MoveNext())
-                                    {
-                                        deltaGroup = deltaGroup.Merge(e.Current);
```

Comment:
The bug happened here, a `NoDelta` null delta marker got passed into this line and `ORDictionary` could not cast it to `DeltaGroup`